imageregistry/v1/types.go

```diff
@@ -303,8 +302,32 @@ type ImageRegistryConfigStorageAzure struct {
 	// object.
 	// +optional
 	CloudName string `json:"cloudName,omitempty"`
+	// networkAccess defines the network access level for the storage account.
+	// "Internal" means the storage account will be private, "External" means
+	// the storage account will be publicly accessible.
+	// Defaults to "External".
+	// +kubebuilder:validation:Enum="Internal";"External"
+	// +kubebuilder:default="External"
+	// +optional
+	NetworkAccess AzureNetworkAccessibility `json:"networkAccess,omitempty"`
+	// vnetName is the name of the vnet the registry operates in.
+	// +optional
+	VNetName string `json:"vnetName,omitempty"`
+	// subnetName is the name of the subnet the registry operates in.
+	// +optional
+	SubnetName string `json:"subnetName,omitempty"`
+	// privateEndpointName is the name of the private endpoint for the registry.
+	// +optional
+	PrivateEndpointName string `json:"privateEndpointName,omitempty"`
```

Comment:
Is there any specific regex for endpoint name? Or, any string is acceptable? Is there a minimal length for the name?

Reply:
added min/max length + regex. 
the regex is based on Azure's resource naming requirements for private endpoints:
```
Alphanumerics, underscores, periods, and hyphens.

Start with alphanumeric. End alphanumeric or underscore.
```